src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/FunctionProbes/FunctionProbesScenario.cs

```diff
@@ -0,0 +1,326 @@
+            new GenericTestMethodSignatures<MyEnum, Uri>().GenericParameters(MyEnum.ValueA, new Uri("https://www.bing.com"), new object());
+
+            Assert.Equal(3, probeProxy.GetProbeInvokeCount(method));
+        }
+
+        public static async Task Test_CaptureValueTypeImplicitThisAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token)
+        {
+            MethodInfo method = typeof(SampleNestedStruct).GetMethod(nameof(SampleNestedStruct.DoWork));
+            SampleNestedStruct nestedStruct = new();
+            await RunTestCaseAsync(probeManager, probeProxy, method, new object[]
+            {
+                5
+            }, nestedStruct, capturedThisObj: null, token);
+        }
+
+        private static async Task Test_CaptureUnsupportedParametersAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token)
+        {
+            MethodInfo method = typeof(StaticTestMethodSignatures).GetMethod(nameof(StaticTestMethodSignatures.RefParam));
+
+            await RunTestCaseCoreAsync(probeManager, probeProxy, method, () =>
+            {
+                int i = 10;
+                StaticTestMethodSignatures.RefParam(ref i);
+
+                return Task.CompletedTask;
+            },
+            new object[]
+            {
+                null
+            },
+            hasImplicitThis: false, capturedThisObj: null, token);
+        }
+
+        private static Task RunAsyncTestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, MethodInfo method, object[] args, object thisObj, CancellationToken token)
+        {
+            return RunAsyncTestCaseAsync(probeManager, probeProxy, method, args, thisObj, thisObj, token);
+        }
+
+        private static Task RunAsyncTestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, MethodInfo method, object[] args, CancellationToken token)
+        {
+            return RunAsyncTestCaseAsync(probeManager, probeProxy, method, args, null, null, token);
+        }
+
+        private static Task RunTestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, MethodInfo method, object[] args, object thisObj, CancellationToken token)
+        {
+            return RunTestCaseAsync(probeManager, probeProxy, method, args, thisObj, thisObj, token);
+        }
+
+        private static Task RunTestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, MethodInfo method, object[] args, CancellationToken token)
+        {
+            return RunTestCaseAsync(probeManager, probeProxy, method, args, null, null, token);
+        }
+
+        private static Task RunAsyncTestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, MethodInfo method, object[] args, object thisObj, object capturedThisObj, CancellationToken token)
```

Comment:
This method and `RunTestCaseAsync` seem to only differ in whether the `MethodInfo` has a `Task` return type or not. Could you combine these, check `method.ReturnType.IsAssignableTo(typeof(Task))`, and then conditionally return its return value or `Task.CompletedTask`?